pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AbstractStructSchema.java

```diff
@@ -80,6 +88,70 @@ public void setSchemaInfoProvider(SchemaInfoProvider schemaInfoProvider) {
         if (reader != null) {
             this.reader.setSchemaInfoProvider(schemaInfoProvider);
         }
+        this.schemaInfoProvider = schemaInfoProvider;
+    }
+
+    public Schema<T> atSchemaVersion(byte[] schemaVersion) throws SchemaSerializationException {
+        Objects.requireNonNull(schemaVersion);
+        if (schemaInfoProvider == null) {
+            // this schema is not downloaded from the registry
+            return this;
```

Comment:
do we care about `supportSchemaVersioning() == false` and/or schemaVersion equality in this case?

Reply:
This is the case in which you have a static Schema (Consumer<GenericRecord> consumer = client.newConsumer(Schema.AVRO(MyPojo.class)), so it is not possible to use the SchemaRegistry and anyway the message is decoded using this static schema.

I am not sure it is worth to add more constraints here in this point.

if I add some other assertion here that case won't work, because the Schema that you pass to the Consumer does not have a "version".